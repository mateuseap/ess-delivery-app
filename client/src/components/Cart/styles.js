import styled from "styled-components";

import { StyledButton } from "../../utils/styles";
import { DEFAULT_RED } from "../App/App.style";

export const CartStyle = styled.div`
  width: 90%;
  margin: auto;
`;

export const TextStyle = styled.h1`
  text-align: center;
  font-weight: bold;
  padding-top: 1.2rem;
  color: black;
`;

export const RestaurantText = styled.div`
  font-size: 2rem;
  padding-left: 10%;
  line-height: 94px;
  padding-top: 0.5rem;
  color: ${DEFAULT_RED};
`;

export const HeaderRow = styled.div`
  font-size: 2rem;
  text-align: center;
  font-weight: bold;
  color: black;
  padding-top: 0.5rem;
  display: flex;
  flex-direction: row;
`;

export const ItemRow = styled.div`
  display: flex;
  flex-direction: row;
  font-size: 1.5rem;
  text-align: center;
  font-weight: bold;
  color: black;
  padding-top: 0.5rem;
  align-items: center;
`;

export const ItemNameHeader = styled.div`
  width: 30%;
  margin-left: 7vw;
`;
export const ItemName = styled.div`
  width: 30%;
`;

export const ItemPrice = styled.div`
  width: 20%;
`;
export const ItemQuantity = styled.div`
  width: 15%;
`;

export const ItemTotal = styled.div`
  margin-left: auto;
  text-align: end;
  width: 15%;
`;

export const AddButton = styled(StyledButton)`
  padding: 0;
  margin-left: 2%;
  margin-right: 2.5%;
  font-size: 2.5rem;
  width: 3.5rem;
  border-radius: 50%;
`;
export const RemoveButton = styled(StyledButton)`
  font-size: 3.4rem;
  padding: 0 7px 6px 5px;
  line-height: 1;
  width: 3.5rem;
  border-radius: 50%;
`;

export const OrderTotalStyle = styled.div`
  color: ${DEFAULT_RED};
  font-size: 1.5rem;
  margin-left: auto;
  width: fit-content;
  padding: 12px 5% 0 50px;
  font-weight: bold;
`;

export const DeliveryFeeNotice = styled.div`
  color: ${DEFAULT_RED};
  font-size: 0.75rem;
  margin-left: auto;
  width: fit-content;
  padding: 5px 5% 20px 50px;
  font-weight: bold;
`;

export const OrderButton = styled(StyledButton)`
  margin-left: auto;
  display: flex;
`;

export const ItemImg = styled.div`
  content: url(${(props) => props.src});
  border-radius: 50px;
  width: 7.5vw;
  height: 8.8vh;
`;

export const RedirectHomeButton = styled(StyledButton)`
  margin: 2% auto 0 auto;
`;
