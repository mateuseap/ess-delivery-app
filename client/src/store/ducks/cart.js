import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  cartRequest: [],
  getCart: [],
  updateCart: ["rest_id", "rest_name", "item", "amountToChange"],
  makeOrder: ["callback"],
  cartSuccess: ["data"],
  cartError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.CART_ERROR]: error,
  [Types.CART_REQUEST]: request,
  [Types.CART_SUCCESS]: success,
});
