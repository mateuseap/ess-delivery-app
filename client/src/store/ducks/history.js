import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  historyRequest: [],
  getHistory: ["dateFilter"],
  postHistory: ["data"],
  historySuccess: ["data"],
  historyError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.HISTORY_ERROR]: error,
  [Types.HISTORY_REQUEST]: request,
  [Types.HISTORY_SUCCESS]: success,
});
