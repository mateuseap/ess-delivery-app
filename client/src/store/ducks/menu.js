import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  menuRequest: [],
  getMenu: ["id"],
  menuSuccess: ["data"],
  menuError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.MENU_ERROR]: error,
  [Types.MENU_REQUEST]: request,
  [Types.MENU_SUCCESS]: success,
});
