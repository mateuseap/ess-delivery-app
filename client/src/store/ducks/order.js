import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  orderRequest: [],
  getOrderDetails: ["id"],
  orderSuccess: ["data"],
  orderError: ["err"],
  cancelOrder: ["id", "callback"],
  cancelOrderWatch: [],
  orderStatusWatchWorker: ["id"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

const cancelOrderWatch = (state) => state.merge({});

export default createReducer(INITIAL_STATE, {
  [Types.ORDER_ERROR]: error,
  [Types.ORDER_REQUEST]: request,
  [Types.ORDER_SUCCESS]: success,
  [Types.CANCEL_ORDER_WATCH]: cancelOrderWatch,
});
