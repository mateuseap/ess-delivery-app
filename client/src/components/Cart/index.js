import React, { Component } from "react";
import {
  CartStyle,
  TextStyle,
  RestaurantText,
  ItemRow,
  HeaderRow,
  OrderButton,
  OrderTotalStyle,
  ItemNameHeader,
  ItemName,
  ItemTotal,
  ItemQuantity,
  ItemPrice,
  RemoveButton,
  AddButton,
  ItemImg,
  RedirectHomeButton,
  DeliveryFeeNotice,
} from "./styles";

import ReactLoading from "react-loading";

import { withRouter } from "../../utils/misc";

import { Link } from "react-router-dom";

import { connect } from "react-redux";
import { Creators as CartCreator } from "../../store/ducks/cart";

import { formatMoney } from "../../utils/misc";
import { deliveryFee } from "../../constants/constants";

class Cart extends Component {
  componentDidMount() {
    this.props.getCart();
  }

  addItem(index) {
    const { cart } = this.props;

    this.props.updateCart(
      cart.data.rest_id,
      cart.data.rest_name,
      cart.data.items[index],
      1
    );
  }

  removeItem(index) {
    const { cart } = this.props;

    this.props.updateCart(
      cart.data.rest_id,
      cart.data.rest_name,
      cart.data.items[index],
      -1
    );
  }

  handleMakeOrder() {
    const callback = (id) => {
      this.props.router.navigate(`/details/${id}`);
    };
    this.props.makeOrder(callback);
  }

  render() {
    const { cart } = this.props;

    return (
      <CartStyle>
        {cart.loading ? (
          <ReactLoading
            type={"spin"}
            style={{
              position: "absolute",
              width: "10vw",
              top: "40%",
              left: "0",
              right: "0",
              margin: "auto",
            }}
          />
        ) : (
          <>
            {cart.data?.items?.length ? (
              <>
                <TextStyle>Seu carrinho de compras</TextStyle>
                <RestaurantText>
                  Restaurante: {cart.data.rest_name}
                </RestaurantText>

                <HeaderRow>
                  <ItemNameHeader>Item</ItemNameHeader>
                  <ItemQuantity>Qtd</ItemQuantity>
                  <ItemPrice>Preço unitário</ItemPrice>
                  <ItemTotal>Total do item</ItemTotal>
                </HeaderRow>

                {cart.data.items?.map((item, index) => (
                  <ItemRow key={item.item_id} name="cartItemRow">
                    <ItemImg src={item.photo}></ItemImg>
                    <ItemName>{item.name}</ItemName>
                    <ItemQuantity>{item.quantity}</ItemQuantity>
                    <ItemPrice>{formatMoney(item.price)}</ItemPrice>
                    <AddButton
                      name="cartAddItemButton"
                      onClick={() => this.addItem(index)}
                    >
                      +
                    </AddButton>
                    <RemoveButton
                      name="cartRemoveItemButton"
                      onClick={() => this.removeItem(index)}
                    >
                      -
                    </RemoveButton>
                    <ItemTotal>
                      {formatMoney(item.quantity * item.price)}
                    </ItemTotal>
                  </ItemRow>
                ))}

                <OrderTotalStyle name="cartTotalPrice">
                  Total: {formatMoney(cart.data.total)}
                </OrderTotalStyle>
                <DeliveryFeeNotice>
                  OBS: É cobrada uma taxa de entrega de
                  {formatMoney(deliveryFee)}
                </DeliveryFeeNotice>
                <OrderButton
                  name="cartMakeOrderButton"
                  onClick={() => this.handleMakeOrder()}
                >
                  Fazer Pedido
                </OrderButton>
              </>
            ) : (
              <>
                <TextStyle name="cartEmptyCartText">
                  Seu carrinho está vazio :(
                  <br />
                  Adicione itens em um de nossos restaurantes!
                </TextStyle>
                <Link
                  name="cartGoHomeButton"
                  to="/home"
                  style={{ textDecoration: "none" }}
                >
                  <RedirectHomeButton>Ver restaurantes</RedirectHomeButton>
                </Link>
              </>
            )}
          </>
        )}
      </CartStyle>
    );
  }
}

const mapStateToProps = ({ cart }) => ({ cart });

export default withRouter(connect(mapStateToProps, { ...CartCreator })(Cart));
