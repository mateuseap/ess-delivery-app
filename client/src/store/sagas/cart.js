import { call, put } from "@redux-saga/core/effects";
import { Creators } from "../ducks/cart";
import api from "../../services/api";
import { toastr } from "react-redux-toastr";

export function* getCart() {
  try {
    yield put(Creators.cartRequest());

    let response = yield call(api.get, "/cart", {});
    yield put(Creators.cartSuccess(response.data));
  } catch (err) {
    yield put(Creators.cartError({ err }));
    toastr.error("Erro ao buscar carrinho");
  }
}

export function* updateCart({ rest_id, rest_name, item, amountToChange }) {
  try {
    yield put(Creators.cartRequest());

    let response = yield call(api.post, "/cart", {
      rest_id,
      rest_name,
      item,
      amountToChange,
    });
    if (response.data) {
      // como so iremos trabalhar com um usuario:
      yield put(Creators.cartSuccess(response.data)); 
      toastr.success("Item adicionado ao carrinho.");
    }
  } catch (err) {
    yield put(Creators.cartError({ err }));
    toastr.error("Erro ao fazer pedido");
  }
}
