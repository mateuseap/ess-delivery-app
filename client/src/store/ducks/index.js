import { combineReducers } from "redux";

import { reducer as toastr } from "react-redux-toastr";

import user from "./user";
import restaurants from "./restaurants";
import history from "./history";
import cart from "./cart";
import menu from "./menu";
import order from "./order";

export default combineReducers({
  user,
  restaurants,
  history,
  cart,
  menu,
  order,
  toastr,
});
