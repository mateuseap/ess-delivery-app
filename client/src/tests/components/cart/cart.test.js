import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Cart from "../../../components/Cart";
import { BrowserRouter } from "react-router-dom";
import { INITIAL_STATE, Types } from "../../../store/ducks/cart";
import createSagaMiddleware from "redux-saga";
import { cartLoadedState } from "../dataUtils";

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";

const sagaMiddleware = createSagaMiddleware();

const mockStore = configureStore([sagaMiddleware]);

Enzyme.configure({ adapter: new Adapter() });

describe("Cart Component Integration Testing", () => {
  it("Renders loading correctly", () => {
    const store = mockStore({ cart: INITIAL_STATE });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Cart />
        </BrowserRouter>
      </Provider>
    );

    const expectedActions = [{ type: Types.GET_CART }];
    expect(store.getActions()).toEqual(expectedActions);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Renders data correctly", () => {
    const store = mockStore({
      cart: cartLoadedState,
    });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Cart />
        </BrowserRouter>
      </Provider>
    );
    const expectedActions = [{ type: Types.GET_CART }];
    expect(store.getActions()).toEqual(expectedActions);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
