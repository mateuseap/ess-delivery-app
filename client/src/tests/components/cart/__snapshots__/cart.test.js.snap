// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component Integration Testing Renders data correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ComponentWithRouterProp>
        <Connect(Cart)
          router={
            Object {
              "location": Object {
                "hash": "",
                "key": "default",
                "pathname": "/",
                "search": "",
                "state": null,
              },
              "navigate": [Function],
              "params": Object {},
            }
          }
        >
          <Cart
            cart={
              Object {
                "data": Object {
                  "items": Array [
                    Object {
                      "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum velit mi, elementum non interdum ut, mattis si",
                      "item_id": 2,
                      "name": "Frango a milanesa",
                      "photo": "https://img.itdg.com.br/tdg/images/recipes/000/021/372/357312/357312_original.jpg?mode=crop&width=710&height=400",
                      "price": 11,
                      "quantity": 1,
                    },
                    Object {
                      "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum velit mi, elementum non interdum ut, mattis si",
                      "item_id": 3,
                      "name": "Cubos de carne ao molho madeira",
                      "photo": "https://www.divenetoalimentos.com.br/images/produtos/picadinho-de-cubos-de-carne-com-molho-congelado.png",
                      "price": 13,
                      "quantity": 1,
                    },
                  ],
                  "rest_id": 0,
                  "rest_name": "Almir quentinhas",
                  "total": 29,
                  "user_id": "1",
                },
                "error": false,
                "loading": false,
              }
            }
            cartError={[Function]}
            cartRequest={[Function]}
            cartSuccess={[Function]}
            getCart={[Function]}
            makeOrder={[Function]}
            router={
              Object {
                "location": Object {
                  "hash": "",
                  "key": "default",
                  "pathname": "/",
                  "search": "",
                  "state": null,
                },
                "navigate": [Function],
                "params": Object {},
              }
            }
            updateCart={[Function]}
          >
            <styled.div>
              <div
                className="sc-dkzDqf emSNjO"
              >
                <styled.h1>
                  <h1
                    className="sc-hKMtZM kDIOHx"
                  >
                    Seu carrinho de compras
                  </h1>
                </styled.h1>
                <styled.div>
                  <div
                    className="sc-eCYdqJ gGLNLP"
                  >
                    Restaurante: 
                    Almir quentinhas
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="sc-jSMfEi jMaENY"
                  >
                    <styled.div>
                      <div
                        className="sc-iBkjds cecdRg"
                      >
                        Item
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-jqUVSM dazhGl"
                      >
                        Qtd
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-papXJ gxnGEq"
                      >
                        Preço unitário
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-kDDrLX buxNlz"
                      >
                        Total do item
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
                <styled.div
                  key="2"
                  name="cartItemRow"
                >
                  <div
                    className="sc-gKXOVf jNUJEn"
                    name="cartItemRow"
                  >
                    <styled.div
                      src="https://img.itdg.com.br/tdg/images/recipes/000/021/372/357312/357312_original.jpg?mode=crop&width=710&height=400"
                    >
                      <div
                        className="sc-hAZoDl jBoOEY"
                        src="https://img.itdg.com.br/tdg/images/recipes/000/021/372/357312/357312_original.jpg?mode=crop&width=710&height=400"
                      />
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-ftvSup lnPzcq"
                      >
                        Frango a milanesa
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-jqUVSM dazhGl"
                      >
                        1
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-papXJ gxnGEq"
                      >
                        R$ 11,00
                      </div>
                    </styled.div>
                    <Styled(styled.button)
                      name="cartAddItemButton"
                      onClick={[Function]}
                    >
                      <button
                        className="sc-gsnTZi sc-iqcoie kRQeln fPfBXj"
                        name="cartAddItemButton"
                        onClick={[Function]}
                      >
                        +
                      </button>
                    </Styled(styled.button)>
                    <Styled(styled.button)
                      name="cartRemoveItemButton"
                      onClick={[Function]}
                    >
                      <button
                        className="sc-gsnTZi sc-crXcEl kRQeln fitzdG"
                        name="cartRemoveItemButton"
                        onClick={[Function]}
                      >
                        -
                      </button>
                    </Styled(styled.button)>
                    <styled.div>
                      <div
                        className="sc-kDDrLX buxNlz"
                      >
                        R$ 11,00
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
                <styled.div
                  key="3"
                  name="cartItemRow"
                >
                  <div
                    className="sc-gKXOVf jNUJEn"
                    name="cartItemRow"
                  >
                    <styled.div
                      src="https://www.divenetoalimentos.com.br/images/produtos/picadinho-de-cubos-de-carne-com-molho-congelado.png"
                    >
                      <div
                        className="sc-hAZoDl hqGKoN"
                        src="https://www.divenetoalimentos.com.br/images/produtos/picadinho-de-cubos-de-carne-com-molho-congelado.png"
                      />
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-ftvSup lnPzcq"
                      >
                        Cubos de carne ao molho madeira
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-jqUVSM dazhGl"
                      >
                        1
                      </div>
                    </styled.div>
                    <styled.div>
                      <div
                        className="sc-papXJ gxnGEq"
                      >
                        R$ 13,00
                      </div>
                    </styled.div>
                    <Styled(styled.button)
                      name="cartAddItemButton"
                      onClick={[Function]}
                    >
                      <button
                        className="sc-gsnTZi sc-iqcoie kRQeln fPfBXj"
                        name="cartAddItemButton"
                        onClick={[Function]}
                      >
                        +
                      </button>
                    </Styled(styled.button)>
                    <Styled(styled.button)
                      name="cartRemoveItemButton"
                      onClick={[Function]}
                    >
                      <button
                        className="sc-gsnTZi sc-crXcEl kRQeln fitzdG"
                        name="cartRemoveItemButton"
                        onClick={[Function]}
                      >
                        -
                      </button>
                    </Styled(styled.button)>
                    <styled.div>
                      <div
                        className="sc-kDDrLX buxNlz"
                      >
                        R$ 13,00
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
                <styled.div
                  name="cartTotalPrice"
                >
                  <div
                    className="sc-evZas blRVCB"
                    name="cartTotalPrice"
                  >
                    Total: 
                    R$ 29,00
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="sc-breuTD lgVUTg"
                  >
                    OBS: É cobrada uma taxa de entrega de
                    R$ 5,00
                  </div>
                </styled.div>
                <Styled(styled.button)
                  name="cartMakeOrderButton"
                  onClick={[Function]}
                >
                  <button
                    className="sc-gsnTZi sc-ksZaOG kRQeln kUOkTm"
                    name="cartMakeOrderButton"
                    onClick={[Function]}
                  >
                    Fazer Pedido
                  </button>
                </Styled(styled.button)>
              </div>
            </styled.div>
          </Cart>
        </Connect(Cart)>
      </ComponentWithRouterProp>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Cart Component Integration Testing Renders loading correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ComponentWithRouterProp>
        <Connect(Cart)
          router={
            Object {
              "location": Object {
                "hash": "",
                "key": "default",
                "pathname": "/",
                "search": "",
                "state": null,
              },
              "navigate": [Function],
              "params": Object {},
            }
          }
        >
          <Cart
            cart={
              Object {
                "data": Array [],
                "error": false,
                "loading": false,
              }
            }
            cartError={[Function]}
            cartRequest={[Function]}
            cartSuccess={[Function]}
            getCart={[Function]}
            makeOrder={[Function]}
            router={
              Object {
                "location": Object {
                  "hash": "",
                  "key": "default",
                  "pathname": "/",
                  "search": "",
                  "state": null,
                },
                "navigate": [Function],
                "params": Object {},
              }
            }
            updateCart={[Function]}
          >
            <styled.div>
              <div
                className="sc-dkzDqf emSNjO"
              >
                <styled.h1
                  name="cartEmptyCartText"
                >
                  <h1
                    className="sc-hKMtZM kDIOHx"
                    name="cartEmptyCartText"
                  >
                    Seu carrinho está vazio :(
                    <br />
                    Adicione itens em um de nossos restaurantes!
                  </h1>
                </styled.h1>
                <Link
                  name="cartGoHomeButton"
                  style={
                    Object {
                      "textDecoration": "none",
                    }
                  }
                  to="/home"
                >
                  <a
                    href="/home"
                    name="cartGoHomeButton"
                    onClick={[Function]}
                    style={
                      Object {
                        "textDecoration": "none",
                      }
                    }
                  >
                    <Styled(styled.button)>
                      <button
                        className="sc-gsnTZi sc-fnykZs kRQeln hikNvC"
                      >
                        Ver restaurantes
                      </button>
                    </Styled(styled.button)>
                  </a>
                </Link>
              </div>
            </styled.div>
          </Cart>
        </Connect(Cart)>
      </ComponentWithRouterProp>
    </Router>
  </BrowserRouter>
</Provider>
`;
